src/store.rs

```diff
@@ -216,53 +64,49 @@ pub trait Store: Clone + Send + Sync {
         Ok(ipld.clone())
     }
 
-    /// Recursively gets a block and all it's references, inserts them into the store,
-    /// pins the root and unpins the old root.
-    ///
-    /// If a block wasn't found it returns a `BlockNotFound` error without modifying the store.
-    async fn sync<'a, I: Into<Cow<'a, Cid>> + Send, N: Into<Cow<'a, Cid>> + Send>(
-        &self,
-        old: Option<I>,
-        new: N,
-    ) -> Result<()>
-    where
-        Ipld: Decode<<Self::Params as StoreParams>::Codecs>,
-    {
-        let mut visited = HashSet::new();
-        let new: Cow<'_, Cid> = new.into();
-        let mut tx = Transaction::new();
-        let mut stack: Vec<Cid> = vec![new.deref().clone()];
-        while let Some(cid) = stack.pop() {
-            if visited.contains(&cid) {
-                continue;
-            }
-            let block = self.get(&cid).await?;
-            for r in block.references()? {
-                stack.push(r);
-            }
-            tx.import(block)?;
-            visited.insert(cid);
-        }
-        tx.update(old, new);
-        self.commit(tx).await
-    }
+    /// Creates an alias for a `Cid`. To alias a block all it's recursive references
+    /// must be in the store. If blocks are missing, they will be fetched from the network. If
+    /// they aren't found, it will return a `BlockNotFound` error.
+    async fn alias<T: AsRef<[u8]> + Send + Sync>(&self, alias: T, cid: Option<&Cid>) -> Result<()>;
 
-    /// Flushes the write buffer.
-    async fn flush(&self) -> Result<()> {
-        Ok(())
-    }
+    /// Resolves an alias for a `Cid`.
+    async fn resolve<T: AsRef<[u8]> + Send + Sync>(&self, alias: T) -> Result<Option<Cid>>;
 }
 
-/// Implemented by ipld storage backends that support aliasing `Cid`s with arbitrary
-/// byte strings.
-#[async_trait]
-pub trait AliasStore: Store {
-    /// Creates an alias for a `Cid` with announces the alias on the public network.
-    async fn alias(&self, alias: &[u8], cid: &Cid) -> Result<()>;
+/// Creates a static alias concatenating the module path with an identifier.
+#[macro_export]
+macro_rules! alias {
+    ($name:ident) => {
+        concat!(module_path!(), "::", stringify!($name))
+    };
+}
 
-    /// Removes an alias for a `Cid`.
-    async fn unalias(&self, alias: &[u8]) -> Result<()>;
+/// Creates a dynamic alias by appending a id.
+pub fn dyn_alias(alias: &'static str, id: u64) -> String {
```

Comment:
'static unnecessary.

Reply:
I think it should be static to ensure it's a compile time string. If you think you know better and want to construct your keys yourself, people are welcome to do so.